### DISCUSSION TITLE
what's the drawback of abusing swr as a client side global storage?

### DISCUSSION THREAD

**Discussion by aboveyunhai** [vercel/swr#2167] [Q&A]
Not really seeking for advices because I wouldn't do that in real project, and just suddenly came up this idea.
but wonder what would happen if I don't use SWR as "data fetching" but as react `useContext()`, what problem it might cause?
The reason I asked that it's because `useContext()` will cause the entire sub-tree re-render if you don't memoize them. Abusing SWR can somehow achieve the same effect without that problem. Of cuz, all similar behaviors are only possible after `mutate` was introduced.

Also the abusing example:  https://codesandbox.io/s/morning-tdd-4o1fkb?file=/src/index.js

**Comment by lfjsa** [comment]
I've pondered the same, but never built a demo to play with the idea.

Hypothetical performance issues aside, `useSWR` runs on promises which makes the store needlessly async, can lead you in to [red/green](https://journal.stuffwithstuff.com/2015/02/01/what-color-is-your-function/) problems, and `const { data }` gets typed as `T | undefined`.
